### PR TITLE
chore: http client healthcheck

### LINKED ISSUE #3315: feature: `client.health`
### Feature request

client to have a health function:
```python
client.health()
await client.async_health()
```

### Motivation

For HTTP, would probably just need to invoke `/readyz`, and for gRPC is to invoke the `Health` rpc from `grpc.health.v1.HealthServicer`.

### Other

_No response_

### PR DESCRIPTION
**Pull request by denyszhak**
## What does this PR address?

Fixes #3315 

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

**Review by aarnphm — COMMENTED** [review]
LGTM. Would kindly request if you can write a quick test for this.

You can write the test under any of the `test/e2e/bento_[grpc|http]_server`.

**Comment by denyszhak** [comment]
> LGTM. Would kindly request if you can write a quick test for this.
> 
> You can write the test under any of the `test/e2e/bento_[grpc|http]_server`.

wdyt if I move simple_service feature on the level above unit and e2e tests? or do you want me to create a separate Service instance for client test

**Comment by aarnphm** [comment]
Lets keep the simple_service fixture in unit, and you can write a simple client test under e2e instead.

No need to create a separate Service instance. The e2e service test will setup everything.

**Comment by denyszhak** [comment]
> Lets keep the simple_service fixture in unit, and you can write a simple client test under e2e instead.
> 
> No need to create a separate Service instance. The e2e service test will setup everything.

Thanks, that makes sense, didn't know the load() API, please check

File: tests/e2e/bento_server_http/tests/test_client.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import pytest
+
+import bentoml
+from bentoml.client import HTTPClient
+
+pytestmark = pytest.mark.asyncio
```
Comment (aarnphm) [inline tests/e2e/bento_server_http/tests/test_client.py:6]: Lets not do this
```suggestion
```

File: tests/e2e/bento_server_http/tests/test_client.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+@pytest.fixture(scope="session")
+def http_client(host: str) -> HTTPClient:
+    service = bentoml.load(bento_identifier="service:svc")
+
+    return HTTPClient(service, f"http://{host}")
+
+
+async def test_async_health(http_client: HTTPClient) -> None:
+    resp = await http_client.async_health()
```
Comment (aarnphm) [inline tests/e2e/bento_server_http/tests/test_client.py:17]:
```suggestion
@pytest.mark.asyncio
async def test_async_health(http_client: HTTPClient) -> None:
    resp = await http_client.async_health()
```

**Review by aarnphm — CHANGES_REQUESTED** [review]
Some finals comments. Otherwise LGTM

File: tests/e2e/bento_server_http/tests/test_client.py (hunk 1 above)
Comment (denyszhak) [reply tests/e2e/bento_server_http/tests/test_client.py]: Of curiosity, what's the reason?

**Review by aarnphm — APPROVED** [review]
Thanks for the contribution.

**Pull request merged**